com.unity.render-pipelines.core/Editor/Converter/RenderPipelineConvertersEditor.cs

```diff
@@ -0,0 +1,436 @@
+    Texture2D kImgPending;
+
+    public VisualTreeAsset m_ConverterEditorAsset;
+    public VisualTreeAsset m_ConverterListAsset;
+    public VisualTreeAsset m_ConverterItem;
+
+    ScrollView m_ScrollView;
+    DropdownField m_ConversionsDropdownField;
+
+    List<RenderPipelineConverter> m_CoreConvertersList = new List<RenderPipelineConverter>();
+    // This list needs to be as long as the amount of converters
+    List<List<ConverterItemDescriptor>> m_ItemsToConvert = new List<List<ConverterItemDescriptor>>();
+    SerializedObject m_SerializedObject;
+
+    List<string> m_ConversionsChoices = new List<string>();
+    // This is a list of Converter States which holds a list of which converter items/assets are active
+    // There is one for each Converter.
+    [SerializeField]
+    List<ConverterState> m_ConverterStates = new List<ConverterState>();
+
+    TypeCache.TypeCollection m_Conversions;
+
+#if RENDER_PIPELINE_CONVERTER
+    [MenuItem("RenderPipelineConverter/RenderPipelineConverter")]
+#endif
+
+    public static void ShowWindow()
+    {
+        RenderPipelineConvertersEditor wnd = GetWindow<RenderPipelineConvertersEditor>();
+        wnd.Show();
+    }
+
+    void OnEnable()
+    {
+        kImgWarn = CoreEditorStyles.iconWarn;
+        kImgHelp = CoreEditorStyles.iconHelp;
+        kImgFail = CoreEditorStyles.iconFail;
+        kImgSuccess = CoreEditorStyles.iconSuccess;
+        kImgPending = CoreEditorStyles.iconPending;
+
+        // This is the drop down choices.
+        m_Conversions = TypeCache.GetTypesDerivedFrom<RenderPipelineConverterContainer>();
+        for (int j = 0; j < m_Conversions.Count; j++)
+        {
+            // Iterate over the conversions
+            RenderPipelineConverterContainer converterContainer = (RenderPipelineConverterContainer)Activator.CreateInstance(m_Conversions[j]);
+            m_ConversionsChoices.Add(converterContainer.name);
+        }
+
+        var converters = TypeCache.GetTypesDerivedFrom<RenderPipelineConverter>();
+        for (int i = 0; i < converters.Count; ++i)
+        {
+            // Iterate over the converters
+            RenderPipelineConverter conv = (RenderPipelineConverter)Activator.CreateInstance(converters[i]);
+            m_CoreConvertersList.Add(conv);
+
+            // Create a new ConvertState which holds the active state of the converter
+            var converterState = new ConverterState
+            {
+                isEnabled = conv.Enabled(),
+                isActive = true,
+                isInitialized = false,
+                items = null,
+                index = i,
+            };
+            m_ConverterStates.Add(converterState);
+
+            // This just creates empty entries in the m_ItemsToConvert.
+            // This list need to have the same amount of entries as the converters
+            List<ConverterItemDescriptor> converterItemInfos = new List<ConverterItemDescriptor>();
+            m_ItemsToConvert.Add(converterItemInfos);
```

Comment:
If I were you, I would create another class storing the `m_CoreConverterList`, the `m_ConverterStates` and the `m_ItemsToConvert`. Something line `ConverterData`( I'm super bad for naming xD)

And I would expose a method

```
void Add(RenderPipelineConverter conv, ConverterState state)
{
  m_CoreConverterList.Add(conv);
  m_ConverterStates.Add(converterState);
  m_ItemsToConvert.Add(new List<ConverterItemDescriptor>)()
}
```

Then this for will be something like:

```
for (int i = 0; i < converters.Count; ++i)
{
           var converterState = new ConverterState
            {
                isEnabled = conv.Enabled(),
                isActive = true,
                isInitialized = false,
                items = null,
                index = i,
            };          
           m _ConverterData.Add(Activator.CreateInstance(converters[i]) as RenderPipelineConverter, converterState);
}

It will be more encapsulated I think

Reply:
Waiting for a next PR rethinking of this.